src/app/components/ListForm/index.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from 'react';
+import Paper from '@mui/material/Paper';
+import InputBase from '@mui/material/InputBase';
+import Divider from '@mui/material/Divider';
+import IconButton from '@mui/material/IconButton';
+import AddIcon from '@mui/icons-material/Add';
+import { useDispatch } from 'react-redux';
+import { addShoppingListItem } from '../../reducers/shoppingListReducer';
+
+export default (): JSX.Element => {
+  const dispatch = useDispatch();
+
+  const [inputValue, setInputValue] = useState<string>('');
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+  };
+
+  const handleAdd = () => {
+    dispatch(addShoppingListItem(inputValue));
```

Comment:
wow, looks quite advanced

Reply:
not really 😬 it is just redux, but thanks!